src/test/java/com/neo4j/docker/neo4jserver/TestBasic.java

```diff
@@ -202,11 +205,48 @@ void testShutsDownCleanly(String signal) throws Exception
     @Test
     void testStartsWhenDebuggingEnabled()
     {
-        try(GenericContainer container = createBasicContainer())
+        try ( GenericContainer container = createBasicContainer() )
         {
             container.withEnv( "NEO4J_DEBUG", "true" );
             container.start();
             Assertions.assertTrue( container.isRunning() );
         }
     }
+
+    @Test
+    void testContainerIsHealthyWhenNeo4jIsListeningAtPort7474()
+    {
+        try ( var container = createBasicContainer() )
+        {
+            container.setWaitStrategy( Wait.forHealthcheck() );
+            container.start();
+
+            Assertions.assertTrue( container.isRunning() );
+            Assertions.assertEquals( "healthy", container.getCurrentContainerInfo().getState().getHealth().getStatus() );
+        }
+    }
+
+    @Test
+    void testContainerIsHealthyWhenListenAddressIsModifiedByUser()
+    {
+        assertContainerWithListenAddressIsHealthy( ":4747" );
+        assertContainerWithListenAddressIsHealthy( "127.0.0.1:7474" );
+        assertContainerWithListenAddressIsHealthy( "localhost:7474" );
+        assertContainerWithListenAddressIsHealthy( "localhost" );
+    }
+
+    void assertContainerWithListenAddressIsHealthy( String listenAddress )
+    {
+        try ( var container = createBasicContainer() )
+        {
+            Map<Setting,Configuration> confNames = Configuration.getConfigurationNameMap();
+            container.withEnv( confNames.get( Setting.HTTP_LISTEN_ADDRESS ).envName, listenAddress );
+
+            container.setWaitStrategy( Wait.forHealthcheck() );
+            container.start();
+
+            Assertions.assertTrue( container.isRunning() );
+            Assertions.assertEquals( "healthy", container.getCurrentContainerInfo().getState().getHealth().getStatus() );
```

Comment:
There is a `container.isHealthy()` function. I don't know if that does what you need

Reply:
I didn't use this as it was not failing my first test before I added any type of health checking. While the container info will not have the container as "healthy" if you have no healthcheck.